File: python-misinformation/vulnerable9.py
from openai import OpenAI
from fastapi import FastAPI
from fastapi.responses import JSONResponse
from pydantic import BaseModel

app = FastAPI()
client = OpenAI()

class Query(BaseModel):
    question: str

@app.post("/fact-check")
async def fact_check(body: Query):
    response = await client.chat.completions.create(
        model="gpt-4o",
        max_tokens=500,
        messages=[{"role": "user", "content": body.question}],
    )
    answer = response.choices[0].message.content
    return JSONResponse({"fact": answer})

File: typescript-misinformation/vulnerable9.ts
import { Controller, Get, Query } from "@nestjs/common";
import OpenAI from "openai";

@Controller("medical")
export class MedicalController {
  private readonly openai = new OpenAI();

  @Get("advice")
  async getAdvice(@Query("symptom") symptom: string): Promise<{ advice: string }> {
    const completion = await this.openai.chat.completions.create({
      model: "gpt-4o",
      messages: [{ role: "user", content: "Medical advice for: " + symptom }],
    });
    return { advice: completion.choices[0].message.content! };
  }
}
